enterprise/coderd/organizations.go

```diff
@@ -161,10 +162,41 @@ func (api *API) deleteOrganization(rw http.ResponseWriter, r *http.Request) {
 		return nil
 	}, nil)
 	if err != nil {
+		orgResourcesRow, queryErr := api.Database.GetOrganizationResourceCountByID(ctx, organization.ID)
+		if queryErr != nil {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error deleting organization.",
+				Detail:  fmt.Sprintf("delete organization: %s", err.Error()),
+			})
+
+			return
+		}
+
+		detailParts := make([]string, 0)
+
+		addDetailPart := func(resource string, count int64) {
+			if count == 1 {
+				detailParts = append(detailParts, fmt.Sprintf("1 %s", resource))
+			} else if count > 1 {
+				detailParts = append(detailParts, fmt.Sprintf("%d %ss", count, resource))
+			}
+		}
+
+		addDetailPart("workspace", orgResourcesRow.WorkspaceCount)
+		addDetailPart("template", orgResourcesRow.TemplateCount)
+
+		// There will always be one member and group so instead we need to check that
+		// the count is greater than one.
+		addDetailPart("member", orgResourcesRow.MemberCount-1)
```

Comment:
I wonder if members should actually prevent deletion. it's not quite as serious as deleting the others which aren't recoverable. workspaces, templates, and groups are all more work to configure, but removing a member doesn't delete the user, and adding them back is quite simple.

I think @Emyrk and I talked about this forever ago and I convinced him that members shouldn't block deletion.

Reply:
Yea, it is a tough one. If we keep members, their roles in that org I think are still fetched on every auth. Maybe we should omit them in that call?


I'd rather be strict today, and loosen it up later

Reply:
Is this something we wan to do in this PR or should we make an issue for it and get to it later?

Reply:
@brettkolodny let's not delete things for the user at this time.

I think we will want to add some "cleanup" button later that the user clicks to explictly delete all the resources required to cleanup an org.